src/runner.rs

```diff
@@ -46,79 +90,147 @@ pub async fn run_test_command(
-                    Err(e) => {
-                        log::warn!("Error opening file: {}", e);
-                        continue;
-                    }
-                };
-                let reader = std::io::BufReader::new(file);
-                let junitxml = match junit_parser::from_reader(reader) {
-                    Ok(junitxml) => junitxml,
-                    Err(e) => {
-                        log::warn!("Error parsing junitxml: {}", e);
-                        continue;
-                    }
-                };
-                for suite in junitxml.suites {
-                    let parent_name = suite.name;
-                    for case in suite.cases {
-                        let failure = case.status.is_failure();
-                        if failure {
-                            let name = case.original_name;
-                            log::debug!("Test failed: {} -> {}", parent_name, name);
-                            failures.push(Test {
-                                parent_name: parent_name.clone(),
-                                name: name.clone(),
-                                class_name: case.classname.clone(),
-                                file: case.file.clone(),
-                            });
-                        }
+            }
+            let file = match std::fs::File::open(&file.original_path) {
+                Ok(file) => file,
+                Err(e) => {
+                    log::warn!("Error opening file: {}", e);
+                    continue;
+                }
+            };
+            let reader = std::io::BufReader::new(file);
+            let junitxml = match junit_parser::from_reader(reader) {
+                Ok(junitxml) => junitxml,
+                Err(e) => {
+                    log::warn!("Error parsing junitxml: {}", e);
+                    continue;
+                }
+            };
+            for suite in junitxml.suites {
+                let parent_name = suite.name;
+                for case in suite.cases {
+                    let failure = case.status.is_failure();
+                    if failure {
+                        let name = case.original_name;
+                        log::debug!("Test failed: {} -> {}", parent_name, name);
+                        failures.push(Test {
+                            parent_name: parent_name.clone(),
+                            name: name.clone(),
+                            class_name: case.classname.clone(),
+                            file: case.file.clone(),
+                        });
                     }
                 }
             }
         }
     }
-    let exit_code = result.code().unwrap_or(EXIT_FAILURE);
-    Ok(RunResult {
+    Ok(failures)
+}
+
+pub async fn run_quarantine(
+    run_result: &RunResult,
+    api_address: &str,
+    token: &str,
+    org_url_slug: &str,
+    repo: &BundleRepo,
+    delay: std::iter::Take<ExponentialBackoff>,
+    no_quarantining: &bool,
```

Comment:
It's unusual to have a `bool` type be a reference because `bool` implements `Copy` allowing it to be copied trivially:
https://doc.rust-lang.org/std/marker/trait.Copy.html

```suggestion
    no_quarantining: bool,
```

Reply:
+1 If you have type which size is less than a machine word, jumping to a reference is more expensive than copy.